packages/x-data-grid-pro/src/hooks/features/columnPinning/useGridColumnPinning.tsx

```diff
@@ -227,7 +227,7 @@ export const useGridColumnPinning = (
   );
 
   const getPinnedColumns = React.useCallback<GridColumnPinningApi['getPinnedColumns']>(() => {
-    return gridPinnedColumnsSelector(apiRef.current.state);
+    return gridPinnedColumnsSelector(apiRef);
   }, [apiRef]);
```

Comment:
I wonder if we should deprecate API methods that only call a selector.

Reply:
Yes please 👍🏻